hasher-matcher-actioner/webapp/src/settings/ThreatExchangeSettingsTab.jsx

```diff
@@ -0,0 +1,244 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+ */
+
+import React, {useEffect, useState} from 'react';
+import {
+  Col,
+  Row,
+  Button,
+  Toast,
+  Form,
+  Spinner,
+  Tooltip,
+  OverlayTrigger,
+  Card,
+} from 'react-bootstrap';
+import PropTypes from 'prop-types';
+import {CopyableTextField} from '../utils/TextFieldsUtils';
+import {
+  fetchAllDatasets,
+  syncAllDatasets,
+  udpateDataset,
+  deleteDataset,
+} from '../Api';
+
+export default function ThreatExchangeSettingsTab() {
+  const [datasets, setDatasets] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const [syncing, setSyncing] = useState(true);
+  const [showToast, setShowToast] = useState(false);
+  const [toastBody, setToastBody] = useState(null);
+  const onPrivacyGroupSave = privacyGroup => {
+    udpateDataset(
```

Comment:
nit: update spelling is wrong.

Reply:
nonblocking:
As @eriksendc's comment yesterday. You could change the state here so that user's can't click the button while the API requests is processing.

Or, you could just update the state assuming the API call would succeed. This is called optimistic updates [1]. In essence, update the state 'before' the API call succeeds. In the then() block, you 'reset' the state and do a failed toast.

1: https://medium.com/@_erikaybar/optimistic-ui-updates-in-react-9e139ffa2e45